streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorContextImpl.java

```diff
@@ -352,6 +358,63 @@ public V fetch(final K key,
         }
     }
 
+    private static class TimestampedWindowStoreReadOnlyDecorator<K, V>
```

Comment:
Since the new classes are static maybe they could go in their own package along with the other pre-existing `decorators`? Probably not on this PR but in a follow-up.

Reply:
I am open to add new packages -- @vvcephei was suggestion this too (for example for RocksDB classes). I would prefer to do this a follow up PRs (it's internal anyway).